libfreerdp/codec/progressive.c

```diff
@@ -460,15 +461,25 @@ static BOOL progressive_allocate_tile_cache(PROGRESSIVE_SURFACE_CONTEXT* surface
 
 	{
 		const RFX_PROGRESSIVE_TILE empty = { 0 };
-
-		void* tmp = realloc(surface->tiles, surface->gridSize * sizeof(RFX_PROGRESSIVE_TILE));
+		void* tmp = realloc(surface->tiles, surface->gridSize * sizeof(RFX_PROGRESSIVE_TILE*));
 		if (!tmp)
 			return FALSE;
 		surface->tilesSize = surface->gridSize;
 		surface->tiles = tmp;
 
+		BOOL allocFailed = FALSE;
 		for (x = oldIndex; x < surface->tilesSize; x++)
-			surface->tiles[x] = empty;
+		{
+			surface->tiles[x] = malloc(sizeof(RFX_PROGRESSIVE_TILE));
```

Comment:
Maybe use `calloc`? or even better, initialize with`NULL` and let the allocation be done by `progressive_tile_allocate`?